app/services/update_stacks_files_service.rb

```diff
@@ -43,7 +43,14 @@ def check_files_in_structural
   def shelve_files
     file_uploads_map.each do |filename, temp_storage_uuid|
       file_path = File.join(Settings.filesystems.transfer, temp_storage_uuid)
-      shelving_path = File.join(stacks_druid_path, filename)
+      shelving_path = stacks_druid_path / filename
+
+      unless shelving_path.to_s.starts_with?(stacks_druid_path.to_s)
```

Comment:
It's (hopefully) a malicious case. As far as I can tell, in the cocina model `filename` is just a string and I don't see anything validating the paths there, so all you'd have to do is somehow generate a filename with `..` or starting with `/`. I don't know if that's possible (maliciously, accidentally or otherwise) in SDR but directory traversal isn't an uncommon problem 🤷

Reply:
I defer to your judgment.

Reply:
I'd file this under the same thing as locking. Hopefully we don't need it, but if we're not 100% certain it can't happen, the cost of the conditional is cheap compared to data loss/corruption.